docker/ghd.py

```diff
@@ -1,384 +1,143 @@
-            await gh.create_deployment_status(deployment_id=args.deployment_id,
-                                              state=args.state,
-                                              environment=args.environment,
-                                              description=args.description)
-    elif cmd == "inspect":
-        argp = argparse.ArgumentParser(description="Inspect deployment state", prog="ghd", epilog=epilogue,
-                                       formatter_class=argparse.ArgumentDefaultsHelpFormatter)
-        argp.add_argument("-r", "--repo", dest="repo",
-                          help="Repository to use, e.g. moneymeets/ghd")
-        argp.add_argument(dest="deployment_id", type=int, nargs="?", help="Deployment ID")
-        args = argp.parse_args(use_argv)
-
-        if args.deployment_id is None and current_deployment is None:
-            print(colorama.Fore.RED + "Missing deployment id" + colorama.Fore.RESET)
-            return
-
-        async with GitHub(repo_path=get_repo_or_fallback(args.repo)) as gh:
-            await gh.inspect(deployment_id=args.deployment_id or int(current_deployment))
-    else:
-        raise RuntimeError(f"Command {cmd} does not exist")
+import click
+import colorama
+from github import DeploymentState, GitHub, get_current_deployment_id, get_current_environment, read_github_event_data
+from util import get_repo_fallback
+
+
+def coroutine(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        return asyncio.run(f(*args, **kwargs))
+
+    return wrapper
+
+
+def click_repo_option():
+    repo_path = get_repo_fallback(read_github_event_data())
+    return click.option("-r", "--repo",
+                        required=repo_path is None,
+                        default=repo_path,
+                        help="Repository to use, e.g. moneymeets/ghd")
+
+
+def click_deployment_id_option():
+    deployment_id = get_current_deployment_id()
+    return click.option("-d", "--deployment-id",
+                        type=int,
+                        required=deployment_id is None,
+                        default=deployment_id,
+                        help="Deployment ID")
+
+
+@click.group()
+def main_group():
+    pass
+
+
+@main_group.command(name="list", short_help="List deployments")
+@click_repo_option()
+@click.option("-v", "--verbose",
+              required=False,
+              is_flag=True,
+              flag_value=True,
+              default=False,
+              help="Print deployment states (slow)")
+@click.option("-l", "--limit",
+              required=False,
+              default=10,
+              help="How many deployments to list")
+@coroutine
+async def cmd_list(repo: str, verbose: bool, limit: int):
+    async with GitHub(repo_path=repo) as gh:
+        await gh.list(limit=limit, verbose=verbose)
+
+
+@main_group.command(name="deploy", short_help="Create new deployment")
+@click_repo_option()
```

Comment:
Wie gestern besprochen werden hier zwei Konzepte vermischt.
Diese Funktion kann von der Kommandozeile über einen Befehl mit entsprechenden Parametern aufgerufen werden. Darüber hinaus wird diese Funktion auch direkt in der Pipeline von GitHub Actions aufgerufen.

`@click_repo_option` liest dabei den Parameter aus der GitHub-Actions-Umgebung und falls er dort nicht vorhanden ist, prüft er den Kommandozeilenparameter.

Aus meiner Sicht müssen die Methoden aufgeteilt werden.
Hier sollten die Methoden implementiert werden, die für die Kommandozeile zur Verfügung stehen. Die Parameter werden dabei immer von den Aufrufparametern gelesen und sind entsprechend als `required` markiert, falls notwendig.

Dann gibt es separate Methoden, die in der Pipeline von GitHub Actions aufgerufen werden. Diese lesen aus der Umgebung die benötigten Parameter und rufen dann das Kommando mit den entsprechenden Parametern auf.

Was meinst Du?